fastapi/routing.py

```diff
@@ -475,7 +477,7 @@ def add_api_route(
             ),
             response_model_exclude_defaults=response_model_exclude_defaults,
             response_model_exclude_none=response_model_exclude_none,
-            include_in_schema=include_in_schema,
+            include_in_schema=include_in_schema if self.include_in_schema else False,
```

Comment:
Using the same variable name makes sense, and I wouldn't do it differently but can be confusing.
```suggestion
            include_in_schema=include_in_schema and self.include_in_schema,
```
Idk if my suggestion makes it smooth to read, no need to use it if you think it doesn't. I'll approve the PR anyway.

Reply:
This means individual routes under a router cannot override the router's decision, if I got that right?

Reply:
@abdusco only if `include_in_schema` on router is set to `False` 

All routes are provided with `include_in_schema=True` by default.

Reply:
@Kludex great suggestion, readability-wise you are right, it will take a moment to understand.

I can commit this suggestion if maintainers think it is better, both are ok with me :smile: